CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/utility/SaveResultsFileWriterNGTest.java

```diff
@@ -53,48 +82,152 @@ public SaveResultsFileWriterNGTest() {
 
     @BeforeMethod
     public void setUpMethod() throws Exception {
+        mockedDataAccessPreferenceKeys = mockStatic(DataAccessPreferenceKeys.class);
+        mockedRecordStoreUtilities = mockStatic(RecordStoreUtilities.class);
+        mockedConstellationLoggerHelper = mockStatic(ConstellationLoggerHelper.class);
     }
 
     @AfterMethod
     public void tearDownMethod() throws Exception {
+        mockedDataAccessPreferenceKeys.close();
+        mockedRecordStoreUtilities.close();
+        mockedConstellationLoggerHelper.close();
     }
 
     /**
-     * Test of generateFilename method, of class SaveResultsFileWriter.
+     * Test of generateFilename method. Ensure that generated filename is of the correct format yyyyMMddTHHmmssSSS.
      */
     @Test
-    public void testGenerateFilenameLength() {
+    public void testGenerateFilename() {
         Plugin plugin = new ExampleClass();
         String filename = SaveResultsFileWriter.generateFilename(plugin, "xml");
-        int expectedFilenameLength = (LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmssSSS")) + "-ExampleClass.xml").length();
-        Assert.assertTrue(filename.length() == expectedFilenameLength);
+        Pattern pattern = Pattern.compile("[0-9]{8}T[0-9]{9}-ExampleClass.xml");
+        Matcher matcher = pattern.matcher(filename.toString());
+        Assert.assertTrue(matcher.matches(), "Testing generated filename matches pattern - yyyyMMddTHHmmssSSS.");
     }
-
+        
     /**
-     * Test of generateFilename method, of class SaveResultsFileWriter.
+     * Test of writeRecordStore method for case when DataAccessPreferenceKeys.getDataAccessResultsDir() returns null.
+     * Expectation is that no attempted writes are made.
      */
     @Test
-    public void testGenerateFilenameSuffix() {
+    public void testWriteRecordStoreNoDataAccessResultsDir() {
         Plugin plugin = new ExampleClass();
-        String filename = SaveResultsFileWriter.generateFilename(plugin, "xml");
-        Assert.assertTrue(filename.endsWith(".xml"));
+        TabularRecordStore tabularRecordStore = new TabularRecordStore();
+                
+        mockedDataAccessPreferenceKeys.when(() -> DataAccessPreferenceKeys.getDataAccessResultsDir()).thenReturn(null);
+        try {
+            SaveResultsFileWriter.writeRecordStore(plugin, tabularRecordStore);
+            Assert.assertTrue(true, "Testing writing record to non existant directory.");
+        } catch (Exception ex) {
+            Assert.assertTrue(false, "Exception is not expected.");
+        }
```

Comment:
I would simply throw the exception here rather than catch it (the test will fail in that case anyway)

Reply:
I went this way for two reasons:
1) generally I like to explictly handle expected exceptions in tests to allow meaningful logging
2) if I dont catch the exception, or throw it, I get compile errros for not catching or declaring the exception.
Happy to change if required.,

Reply:
@serpens24 So the reason I suggest throwing rather than catching is that is what we've been doing in other unit tests for other classes. Generally speaking, I think it looks neater to throw and prefer not to catch unless a test does something more than just assert it got there after the catch happens (e.g. like what you've done with your other test).

I can understand wanting to have consistency in handling across the class though so I'm happy to leave it to you as to whether you want to change it or leave it

Reply:
I've opted to add the throws Exception call and go with your suggestion if its being done like that elsewhere. Its probably always an ongoing debate :)